### ISSUE TITLE
Reach out to Kotlin plugin to reduce its impact to task configuration avoidance

### ISSUE THREAD

**Issue by big-guy** [gradle/gradle-native#732]
The Kotlin plugin reaches deep into the task container and resolves many tasks by name or type:
https://github.com/JetBrains/kotlin/blob/master/libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinPlugin.kt

To name a few:
- clean
- Each source set's classes task
- jar tasks
- compilation tasks

This means that any project that uses this plugin will see little/no gain from the new APIs.

**Comment by oehme** [comment]
I'm concerned that asking them to focus on lazy tasks will divert attention from the much bigger issues they currently have, like increasing build times by 3x. I'd rather have them focus on that first. Slightly higher configuration time is currently the smallest worry of Kotlin users.

**Comment by lacasseio** [comment]
@big-guy What do you say we close this and not consider this as a blocker. The alternative would be to reach in ourselves in the plugin and provide PRs to fix the couple lazy task issues. However, I agree that increasing build time by 3x would be a better investment of their time. We could frame it more like please put this (lazy task conversion) in your backlog and get to it when you can.

**Comment by lacasseio** [comment]
That would be nice of you. Do you want to keep this issue around or will you make a copy so we can close it?

**Comment by oehme** [comment]
Just assign this one to me and I'll take care of it.

**Comment by big-guy** [comment]
@oehme I agree that this isn't the most important thing for them to do. Just improving the code in general may incidentally fix some of these issues.

@lacasseio I think we just park this out of our iteration, but on our TODO list that we can scrub later.
